chia/wallet/wallet_state_manager.py

```diff
@@ -499,21 +506,29 @@ async def does_coin_belong_to_wallet(self, coin: Coin, wallet_id: int) -> bool:
 
         return False
 
+    async def get_confirmed_balance_for_wallet_already_locked(self, wallet_id: int) -> uint128:
+        # This is a workaround to be able to call la locking operation when already locked
+        # for example, in the create method of DID wallet
+        assert self.lock.locked() is False
+        unspent_coin_records = await self.coin_store.get_unspent_coins_for_wallet(wallet_id)
+        return get_balance_from_coin_records(unspent_coin_records)
+
     async def get_confirmed_balance_for_wallet(
         self, wallet_id: int, unspent_coin_records: Optional[Set[WalletCoinRecord]] = None
     ) -> uint128:
         """
         Returns the confirmed balance, including coinbase rewards that are not spendable.
         """
-        # lock only if unspent_coin_records is None
+        # lock only if unspent_coin_records is None.
+        # This API should change so that get_balance_from_coin_records is called for Set[WalletCoinRecord]
+        # and this method is called only for the unspent_coin_records==None case.
```

Comment:
I would think `unspent_coin_records` should be removed as a parameter, and any caller that wants to pass it in should just call `get_balance_from_coin_records()` instead. But this patch should be kept minimal